espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/KReflectionUtils.kt

```diff
@@ -0,0 +1,67 @@
+package io.appium.espressoserver.lib.helpers
+
+import io.appium.espressoserver.lib.handlers.exceptions.AppiumException
+import java.lang.IllegalArgumentException
+import java.lang.reflect.InvocationTargetException
+import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
+import kotlin.reflect.full.createType
+import kotlin.reflect.full.functions
+import kotlin.reflect.full.memberFunctions
+import kotlin.reflect.jvm.javaType
+
+object KReflectionUtils {
+
+    fun invokeMethod(functions: Collection<KFunction<*>>, methodName: String, vararg providedParams: Any): Any? {
+        val treatedParams = providedParams.clone().toMutableList()
+        for (func in functions) {
+            // Look for function names that match provided methodName
+            if (func.name == methodName) {
+
+                // If the length of func parameters provided isn't same as expected, go to next.
+                val funcParams = func.parameters
+                if (funcParams.size != providedParams.size) {
+                    continue
+                }
+
+                // Look through function parameters and do an enum hack to translate strings to enums
+                for (index in 0 until funcParams.size) {
```

Comment:
how about using https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/for-each-indexed.html ?